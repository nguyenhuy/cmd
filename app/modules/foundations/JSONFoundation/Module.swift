Target.module(
  name: "JSONFoundation",
  dependencies: [],
  testDependencies: [])
