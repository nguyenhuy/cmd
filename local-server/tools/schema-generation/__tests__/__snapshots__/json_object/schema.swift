// swiftformat:disable all
// This file is generated from ./local-server/src/server/schemas/schemaFile.ts by `yarn export-schema-swift`.
// Do not edit it manually.

import Foundation
import JSONFoundation

extension Schema {
  public struct Wrapper: Codable, Sendable {
    public let properties: JSON
  
    private enum CodingKeys: String, CodingKey {
      case properties = "properties"
    }
  
    public init(
        properties: JSON
    ) {
      self.properties = properties
    }
  
    public init(from decoder: Decoder) throws {
      let container = try decoder.container(keyedBy: CodingKeys.self)
      properties = try container.decode(JSON.self, forKey: .properties)
    }
  
    public func encode(to encoder: Encoder) throws {
      var container = encoder.container(keyedBy: CodingKeys.self)
      try container.encode(properties, forKey: .properties)
    }
  }}
