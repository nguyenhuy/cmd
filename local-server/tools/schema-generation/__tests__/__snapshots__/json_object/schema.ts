export interface Wrapper {
  properties: Record<string, unknown>;
}
