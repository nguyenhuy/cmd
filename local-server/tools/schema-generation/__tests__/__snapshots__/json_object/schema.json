{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "$ref": "#/definitions/Wrapper",
  "definitions": {
    "Wrapper": {
      "type": "object",
      "properties": {
        "properties": {
          "type": "object",
          "additionalProperties": {}
        }
      },
      "required": [
        "properties"
      ],
      "additionalProperties": false
    }
  }
}